parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -0,0 +1,283 @@
+
+    def create_headernode(self,header):
+        """ create the node for the header """
+        query = """
+                CREATE (n:TEXT {id: '__header__' })
+                SET n.preceding_lines= $header
+                SET n.src_position= 1
+            """
+        self.session.run(query,header=header)
+
+    def create_node(self,data):
+        """ run the query to create the node with data dictionary """
+        position_query = """
+            SET n.is_before = $is_before
+            SET n.preceding_lines= $comment
+            SET n.src_position= $src_position
+        """
+        entry_query=""
+        if data['id'] == '__footer__':
+            id_query = " CREATE (n:TEXT {id: $id }) \n "
+        elif data['id'].startswith('synonyms'):
+            id_query = " CREATE (n:SYNONYMS {id: $id }) \n "
+        elif data['id'].startswith('stopwords') :
+            id_query = " CREATE (n:STOPWORDS {id: $id }) \n "
+        else :
+            id_query = " CREATE (n:ENTRY {id: $id , main_language : $main_language}) \n "
+            if data['parent_tag'] : entry_query += " SET n.parents = $parent_tag \n"
+            for key in data :
+                if key.startswith('prop_') :
+                    entry_query += " SET n." + key + " = $" + key + '\n'
+
+        for key in data :
+            if key.startswith('tags_') :
+                entry_query += " SET n." + key + " = $" + key + '\n'
+
+        query = id_query + entry_query + position_query
+        self.session.run(query,data,is_before=self.is_before,)
+
+    def normalized_filename(self,filename):
+        """ add the .txt extension if it is missing in the filename """
+        return filename + ( '.txt' if (len(filename)<4 or filename[-4:]!=".txt") else '')
+
+    def file_iter(self,filename,start=0):
+        """generator to get the file line by line"""
+        with open(filename,"r",encoding="utf8") as file:
+            for line_number,line in enumerate(file):
+                if line_number < start :
+                    continue
+                # sanitizing
+                #remove any space characters at end of line
+                line = line.rstrip()
+                #replace ’ (typographique quote) to simple quote '
+                line = line.replace("’", "'")
+                # replace commas that have no space around by a lower comma character 
+                # and do the same for escaped comma (preceded by a \) 
+                # (to distinguish them from commas acting as tags separators)
+                line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+                line = re.sub(r"\\,", "\\‚", line)
+                #removes parenthesis for roman numeral
+                line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+                yield line_number,line
+        yield line_number,"" #to end the last entry if not ended
+
+    def normalizing(self,line,lang="default"):
```

Comment:
@BryanH01 this whole function could be supplemented with some spaces to make it more readable.